### ISSUE TITLE
Allow installed PWA to run on OS login

### ISSUE THREAD

**Issue by laukstein** [w3c/manifest#845]
I think we should standardize it before its wrong approaches spreads out.
Details in https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/RunOnLogin/Explainer.md and https://bugs.chromium.org/p/chromium/issues/detail?id=897302

![pwa-dialog](https://user-images.githubusercontent.com/183966/72961965-9834d780-3dbb-11ea-98aa-92c6dac8eb8f.png)

**Comment by marcoscaceres** [comment]
@aarongustafson, is that something Microsoft if planning to run through the WICG?

**Comment by marcoscaceres** [comment]
I wonder if this would just be an OS specific thing instead of a manifest thing? Like, with MacOS apps, the user can set this with the following menu: 

<img width="511" alt="Screenshot 2020-04-02 10 37 25" src="https://user-images.githubusercontent.com/870154/78196382-f4a82780-74cd-11ea-889d-32e3ee07a299.png">

**Comment by reillyeon** [comment]
It is true most/all operating systems provide a way for the user to enable launching an application on login but I think the intuition in Microsoft's proposal is that users expect to find these options within the application itself and are unfamiliar with the OS-level settings. The proposed manifest property allows the application to hint to the browser that this is an application for which the user is likely to want this option surfaced more prominently.

**Comment by marcoscaceres** [comment]
> The proposed manifest property allows the application to hint to the browser that this is an application for which the user is likely to want this option surfaced more prominently.

Understood. My worry is that this will become really spammy as everyone will want this... kinda the reason that OSs hide this option, and keeps the user in control.

**Comment by reillyeon** [comment]
> My worry is that this will become really spammy as everyone will want this... kinda the reason that OSs hide this option, and keeps the user in control.

I agree with the concern that this could be spammy. The user agent retains the ability to keep the user in control by not granting this capability whenever it is requested. That is, the checkbox default is unchecked.

As I noted in MicrosoftEdge/MSEdgeExplainers#158 I wonder how much the ability to receive push notifications on login, as is possible already on Android and Chrome OS, would resolve developer needs. Combining that with better in-browser controls for launching installed PWAs on login may obviate the desire for this manifest properly. It is worth noting that Chromium does not currently include installed PWAs in its "session restore" functionality ([issue 938759](https://bugs.chromium.org/p/chromium/issues/detail?id=938759)) and this may contribute to developer desire for this explicit control.

**Comment by aarongustafson** [comment]
> @aarongustafson, is that something Microsoft if planning to run through the WICG?

I believe so. @jungkees & @carlosfriasand are the owners.

**Comment by jungkees** [comment]
Apologies for the late response to the thread. And thanks @reillyeon for having provided the context.

I just posted to the WICG discourse to get this concept be discussed there. We totally get the concerns discussed in this thread and consider the capability a [powerful feature](https://w3c.github.io/permissions/#powerful-feature). We've updated [the explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/RunOnLogin/Explainer.md) with JS APIs that could be used to implement an in-app settings to enable/disable the capability.

We still retain an option to use the web app manifest hinging on the [Install Time Permissions Prompt](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/master/InstallTimePermissionsPrompt/Explainer.md) work. However, it'd be great if we could discuss on our proposal in WICG at this point.
